dashboard/modules/reporter/reporter_agent.py

```diff
@@ -610,28 +604,13 @@ async def _perform_iteration(self, publish):
 
     async def run(self, server):
         reporter_pb2_grpc.add_ReporterServiceServicer_to_server(self, server)
-        if gcs_pubsub_enabled():
-            gcs_addr = self._dashboard_agent.gcs_address
-            if gcs_addr is None:
-                aioredis_client = await aioredis.create_redis_pool(
-                    address=self._dashboard_agent.redis_address,
-                    password=self._dashboard_agent.redis_password,
-                )
-                gcs_addr = await aioredis_client.get("GcsServerAddress")
-                gcs_addr = gcs_addr.decode()
-            publisher = GcsAioPublisher(address=gcs_addr)
-
-            async def publish(key: str, data: str):
-                await publisher.publish_resource_usage(key, data)
 
-        else:
-            aioredis_client = await aioredis.create_redis_pool(
-                address=self._dashboard_agent.redis_address,
-                password=self._dashboard_agent.redis_password,
-            )
+        gcs_addr = self._dashboard_agent.gcs_address
+        assert gcs_addr is not None
+        publisher = GcsAioPublisher(address=gcs_addr)
 
-            async def publish(key: str, data: str):
-                await aioredis_client.publish(key, data)
+        async def publish(key: str, data: str):
```

Comment:
Maybe this can be merged into `_perform_iteration()`, instead of having to pass into it.